src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -193,9 +252,31 @@ protected void handleMessageInternal(final Message<?> message) throws Exception
 			headers = new RecordHeaders();
 			this.headerMapper.fromHeaders(message.getHeaders(), headers);
 		}
-		ProducerRecord<K, V> producerRecord = new ProducerRecord<K, V>(topic, partitionId, timestamp, (K) messageKey,
-				payload, headers);
-		ListenableFuture<?> future = this.kafkaTemplate.send(producerRecord);
+		final ProducerRecord<K, V> producerRecord = new ProducerRecord<K, V>(topic, partitionId, timestamp,
+				(K) messageKey, payload, headers);
+		ListenableFuture<SendResult<K, V>> future = this.kafkaTemplate.send(producerRecord);
+		if (getSendFailureChannel() != null || getOutputChannel() != null) {
+			future.addCallback(new ListenableFutureCallback<SendResult<K, V>>() {
+
+				@Override
+				public void onSuccess(SendResult<K, V> result) {
+					if (getOutputChannel() != null) {
+						KafkaProducerMessageHandler.this.messagingTemplate.send(getOutputChannel(),
+								getMessageBuilderFactory().withPayload(result.getRecordMetadata()).build());
```

Comment:
No, I mean leave incoming `payload` as is but add `RecordMetadata` as a header for future analysis:
```
getMessageBuilderFactory()
            .fromMessage(message)
            .setHeader("kafka_recordMetadata", recordMetadata)
            .build()
```

The problem that this component isn't reply producer. There is just nothing to reply, but for some reason we really would like to track send ACKs in our system. Therefore we need to correlate request to the ACK somehow.  IMO the best way to keep `requestMessage` as is but enhance with the ACK info.

I'm may missing something, but at least that is how I feel the feature.